imagebuildah/stage_executor.go

```diff
@@ -1175,6 +1199,13 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 		// image, run the step so that we can check if the result
 		// matches a cache.
 		if cacheID == "" {
+			// Note: In theory we should never ever hit this branch now
+			// since theoretically only `COPY` and `ADD` should  fall in
+			// this branch but we already process them with execption cases
+			// above to save cycles. But still leave this branch here so if
+			// if we missed anything then we still end up giving our best
+			// effort to find a cache hit.
+
```

Comment:
I am fine to drop this entire branch. This is not needed anyways as i have written in comments.

Reply:
When would instructions other than ADD and COPY be processed, then?

Reply:
Where is RUN going to happen if this is completely dropped?

Reply:
My bad wrong wordings , my intent was that if there was a `cache` hit `RUN` would never hit this branch afaics in case of valid cache hits `RUN` should always get resolved by the first attempt to find the cache. But in case there is no cache hits this branch is still useful.

Reply:
In recent push this block would only match for cases which are not `COPY` or `ADD` and missed the cache lookup in the first pass.

Reply:
Closing this as resolved.